openwisp_monitoring/monitoring/tasks.py

```diff
@@ -1,13 +1,35 @@
+from datetime import datetime
+
+import pytz
 from celery import shared_task
+from django.core.exceptions import ObjectDoesNotExist
+from swapper import load_model
 
 from ..db import timeseries_db
 from ..db.exceptions import TimeseriesWriteException
 from .settings import RETRY_OPTIONS
 
 
 @shared_task(bind=True, autoretry_for=(TimeseriesWriteException,), **RETRY_OPTIONS)
-def timeseries_write(self, name, values, **kwargs):
+def timeseries_write(self, name, values, metric_pk=None, threshold_dict=None, **kwargs):
     """
     write with exponential backoff on a failure
     """
     timeseries_db.write(name, values, **kwargs)
+    if not metric_pk or not threshold_dict:
+        return
+    try:
+        metric = load_model('monitoring', 'Metric').objects.get(pk=metric_pk)
+        time = threshold_dict.pop('time')
+        if isinstance(time, str):
+            time = (
+                datetime.strptime(time, '%Y-%m-%dT%H:%M:%S.%fZ')
+                if 'Z' in time
+                else datetime.strptime(time, '%Y-%m-%dT%H:%M:%S.%f')
+            )
+            time = pytz.utc.localize(time)
```

Comment:
is there a simpler way to do this? Isn't the time we get always supposed to be in UTC?

Also, I think it would be better to move it to `check_threshold`
I wold simply call `metric.check_threshold(**check_threshold_kwargs)` and let `check_threshold` figure out what it has to do.